### PR TITLE
Diagnostic API fixes

### PR DESCRIPTION
**Pull request by nnethercote**
Some improvements to diagnostic APIs: improve some naming, use shortcuts in more places, and add a couple of missing methods.

r? @compiler-errors

### CODE REVIEW

**Comment by nnethercote** [comment]
@oli-obk: The [Rename consuming chaining methods on DiagnosticBuilder](https://github.com/rust-lang/rust/pull/119751/commits/61905593a589ae98d1e67498b0b8a0e666f25ae1) commit renames the `foo_mv` methods as `with_foo`, because that's an established naming convention for such methods.

File: compiler/rustc_errors/src/diagnostic_builder.rs [hunk 1]
```diff
@@ -173,26 +173,26 @@ impl EmissionGuarantee for rustc_span::fatal_error::FatalError {
...
 ///   It is mostly used in a chained fashion when producing a new diagnostic,
-///   e.g. `let err = struct_err(msg).code_mv(code)`, or when emitting a new
-///   diagnostic , e.g. `struct_err(msg).code_mv(code).emit()`.
+///   e.g. `let err = struct_err(msg).with_code(code)`, or when emitting a new
+///   diagnostic , e.g. `struct_err(msg).with_code(code).emit()`.
 ///
 /// Although the latter method can be used to modify an existing diagnostic,
-/// e.g. `err = err.code_mv(code)`, this should be avoided because the former
+/// e.g. `err = err.with_code(code)`, this should be avoided because the former
 /// method give shorter code, e.g. `err.code(code)`.
```
Comment (RalfJung) [inline compiler/rustc_errors/src/diagnostic_builder.rs:183]:
```suggestion
/// method gives shorter code, e.g. `err.code(code)`.
```

Comment (nnethercote) [reply compiler/rustc_errors/src/diagnostic_builder.rs]: Fixed, thanks.

**Comment by compiler-errors** [comment]
r? oli-obk

kind of busy today so i want to offload some currently assigned reviews

**Pull request merged**